### PR TITLE
fix splitting metadata key-value substring with more than one '=' sign

### PR DESCRIPTION
**Pull request by malafeev**
cc @carlosalberto

### CODE REVIEW

**Comment by malafeev** [comment]
@anuraaga I modified existing test to check values with more than one `=`

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java [hunk 1]
```diff
@@ -291,7 +291,7 @@ protected Builder fromConfigMap(
       String metadataValue = getStringProperty(KEY_METADATA, configMap);
       if (metadataValue != null) {
         for (String keyValueString : Splitter.on(';').split(metadataValue)) {
-          final List<String> keyValue = Splitter.on('=').splitToList(keyValueString);
+          final List<String> keyValue = Splitter.on('=').limit(2).splitToList(keyValueString);
```
Comment (anuraaga) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java:294]: Should we also add `trimResults` `omitEmptyStrings`? These are HTTP headers so at least keys cannot have spaces or be empty. I'm less sure about trimming the values or the key only, but I think usually a trailing space on a value will be a bug not a feature.

Comment (malafeev) [reply exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java]: added

File: exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java [hunk 2]
```diff
@@ -291,7 +291,8 @@ protected Builder fromConfigMap(
       String metadataValue = getStringProperty(KEY_METADATA, configMap);
       if (metadataValue != null) {
         for (String keyValueString : Splitter.on(';').split(metadataValue)) {
-          final List<String> keyValue = Splitter.on('=').splitToList(keyValueString);
+          final List<String> keyValue = Splitter.on('=').limit(2).trimResults().omitEmptyStrings()
```
Comment (anuraaga) [inline exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java:294]: Thanks! If you can add two more test cases, one with trimmed results and one with empty strings, that'd be great.

Comment (malafeev) [reply exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcSpanExporter.java]: added
